pkg/driver_sx126x/contrib/driver_sx126x_hal.c

```diff
@@ -59,26 +113,77 @@ sx126x_hal_status_t sx126x_hal_read(const void *context,
 {
     sx126x_t *dev = (sx126x_t *)context;
 
-    /* wait for the device to not be busy anymore */
-    while (gpio_read(dev->params->busy_pin)) {}
+    if (sx126x_is_stm32wl(dev)) {
+#if IS_USED(MODULE_SX126X_STM32WL)
+        sx126x_radio_wait_until_ready(dev);
+
+        spi_acquire(dev->params->spi, SPI_CS_UNDEF, SX126X_SPI_MODE, SX126X_SPI_SPEED);
+        /* Pull NSS low */
+        PWR->SUBGHZSPICR &= ~PWR_SUBGHZSPICR_NSS;
+        spi_transfer_bytes(dev->params->spi, SPI_CS_UNDEF, true, command, NULL, \
+                           command_length);
+        spi_transfer_bytes(dev->params->spi, SPI_CS_UNDEF, false, NULL, data, \
+                           data_length);
+        /* Pull NSS high */
+        PWR->SUBGHZSPICR |= PWR_SUBGHZSPICR_NSS;
+#endif
+    }
+    else {
+        /* wait for the device to not be busy anymore */
+        while (gpio_read(dev->params->busy_pin)) {}
+
+        spi_acquire(dev->params->spi, SPI_CS_UNDEF, SX126X_SPI_MODE, SX126X_SPI_SPEED);
+        spi_transfer_bytes(dev->params->spi, dev->params->nss_pin, true, \
+                           command, NULL, command_length);
+        spi_transfer_bytes(dev->params->spi, dev->params->nss_pin, false, \
+                           NULL, data, data_length);
+    }
 
-    spi_acquire(dev->params->spi, SPI_CS_UNDEF, SX126X_SPI_MODE, SX126X_SPI_SPEED);
-    spi_transfer_bytes(dev->params->spi, dev->params->nss_pin, true, command, NULL, command_length);
-    spi_transfer_bytes(dev->params->spi, dev->params->nss_pin, false, NULL, data, data_length);
     spi_release(dev->params->spi);
     return 0;
 }
 
 sx126x_hal_status_t sx126x_hal_reset(const void *context)
 {
-    DEBUG("[sx126x_hal] reset\n");
     sx126x_t *dev = (sx126x_t *)context;
 
-    gpio_set(dev->params->reset_pin);
-    gpio_clear(dev->params->reset_pin);
-    /* it takes 100us for the radio to be ready after reset */
-    ztimer_sleep(ZTIMER_USEC, 100);
-    gpio_set(dev->params->reset_pin);
+    if (sx126x_is_stm32wl(dev)) {
+#if IS_USED(MODULE_SX126X_STM32WL)
+        /* Reset Radio */
+        RCC->CSR |= RCC_CSR_RFRST;
+        /* Clear radio reset */
+        RCC->CSR &= ~RCC_CSR_RFRST;
+        ztimer_sleep(ZTIMER_USEC, 100);
+        /* Wait while reset is done */
+        while((RCC->CSR & RCC_CSR_RFRSTF) != 0UL) {}
```

Comment:
Can you uncristify?

Reply:
I think I will do it once in a separate commit once all fixups are done :)